src/vm/dllimportcallback.cpp

```diff
@@ -1416,6 +1416,7 @@ VOID UMThunkMarshInfo::RunTimeInit()
     {
         // caller pop
         m_cbRetPop = 0;
+        m_cbActualArgSize += numRegistersUsed * STACK_ELEM_SIZE;
```

Comment:
@lucenticus looking at the code and thinking about it more, it seems the problem is caused by the fact that when we get the MetaSig for iteration over the arguments, we use the pMD. But that's right only if we are not generating any stub, which seems to be the case of the NativeCallable attribute.
However, in other cases, I think we should create the MetaSig from the pStubMD instead. I may be wrong, but I think we can just use `MetaSig sig((pStubMD != NULL )? pStubMD : pMD))`.